src/radical/pilot/utils/misc.py

```diff
@@ -219,5 +219,112 @@ def get_resource_job_url(resource: str,
     return ru.Url(rcfg.schemas[schema]['job_manager_endpoint'])
 
 
+# ------------------------------------------------------------------------------
+#
+def convert_slots_to_new(slots, log=None):
+
+    from ..resource_config import Slot, RO
+
+    if not slots:
+        return slots
+
+    new_slots = list()
+    for slot in slots:
+
+        # check slot type: only new slots have a version field
+        if slot.get('version'):
+            new_slots.append(slot)
+            continue
```

Comment:
isn't it if version is set (`slot['version'] = 'old'` for JSRUN) then it is old, but if it is not set (`0` is default value) then it is new

Reply:
-> `if not slot.get('version'):`

Reply:
Version is not set for slots which come from the test suite's jsrun tests, and this is when this code is kicking in.  Slots from the scheduler always have `version` set and can be used as is.

This will all go away as soon as we retire jsrun.

Reply:
> Version is not set for slots which come from the test suite's jsrun tests, and this is when this code is kicking in. Slots from the scheduler always have version set and can be used as is.

yeah, my point was that it is opposite 
https://github.com/radical-cybertools/radical.pilot/blob/47c0ea5a61075e70c8ccfaf59426583f9a87641b/tests/unit_tests/test_lm/test_jsrun.py#L135-L136

and for all Slots default value for `version` is `0` (which is treated in these statements as not set)